shared/src/main/java/com/couchbase/lite/Database.java

```diff
@@ -760,81 +760,59 @@ C4Document read(String docID, boolean mustExist)
+                    if (!otherDoc.selectConflictingRevision())
+                        return false;
 
-                // Read the common ancestor revision (if it's available):
-                Document baseDoc = new Document(this, docID, true, true);
-                if (!baseDoc.selectCommonAncestor(doc, otherDoc) || baseDoc.getData() == null)
-                    baseDoc = null;
+                    // Read the common ancestor revision (if it's available):
+                    baseDoc = new Document(this, docID, true);
+                    if (!baseDoc.selectCommonAncestor(doc, otherDoc) || baseDoc.toMap() == null)
+                        baseDoc = null;
 
-                // Call the conflict resolver:
-                Document resolved;
-                if (otherDoc.isDeleted()) {
-                    resolved = doc;
-                } else if (doc.isDeleted()) {
-                    resolved = otherDoc;
-                } else {
-                    if (resolver == null)
-                        resolver = effectiveConflictResolver();
-                    Conflict conflict = new Conflict(doc, otherDoc, baseDoc);
-                    Log.i(TAG, "Resolving doc '%s' with %s (mine=%s, theirs=%s, base=%s)",
-                            docID,
-                            resolver != null ? resolver.getClass().getSimpleName() : "null",
-                            doc != null ? doc.getRevID() : "null",
-                            otherDoc != null ? otherDoc.getRevID() : "null",
-                            baseDoc != null ? baseDoc.getRevID() : "null");
-                    resolved = resolver.resolve(conflict);
-                    if (resolved == null)
-                        throw new CouchbaseLiteException(LiteCoreDomain, kC4ErrorConflict);
+                    commit = true;
+                } finally {
+                    endTransaction(commit);
                 }
 
-                // Figure out what revision to delete and what if anything to add:
-                String winningRevID;
-                String losingRevID;
-                byte[] mergedBody = null;
-                if (resolved == otherDoc) {
-                    winningRevID = otherDoc.getRevID();
-                    losingRevID = doc.getRevID();
-                } else {
-                    winningRevID = doc.getRevID();
-                    losingRevID = otherDoc.getRevID();
-                    if (resolved != doc) {
-                        resolved.setDatabase(this);
-                        try {
-                            mergedBody = resolved.encode();
-                        } catch (LiteCoreException e) {
-                            throw LiteCoreBridge.convertException(e);
-                        }
-                        if (mergedBody == null)
-                            return false;
-                    }
-                }
+                // Call the conflict resolver:
+                if (resolver == null)
```

Comment:
resolver should never be null.

Reply:
this `resolver` is passed from `ReplicatorConfiguration`. `ReplicatorConfiguration.conflictResolver` could be null if the developer does not set it.